examples/tinywl/surfacewrapper.cpp

```diff
@@ -154,6 +154,14 @@ void SurfaceWrapper::setParent(QQuickItem *item)
 void SurfaceWrapper::setActivate(bool activate)
 {
     m_shellSurface->setActivate(activate);
+    if (type() == SurfaceWrapper::Type::XWayland) {
+        auto xwaylandSurface = qobject_cast<WXWaylandSurface *>(m_shellSurface);
+        if (activate)
+            xwaylandSurface->restack(nullptr, WXWaylandSurface::XCB_STACK_MODE_ABOVE);
```

Comment:
You should call `restack`  in `SurfaceWrapper::stackAfter` or `SurfaceWrapper::stackBefore` is not in here.

Reply:
add in Helper::setActivatedSurface